src/XRP/model/xrp-currency.ts

```diff
@@ -14,12 +15,26 @@ export default class XrpCurrency {
    * @see https://github.com/ripple/rippled/blob/develop/src/ripple/proto/org/xrpl/rpc/v1/amount.proto#L41
    */
   public static from(currency: Currency): XrpCurrency {
-    return new XrpCurrency(currency.getName(), currency.getCode_asU8())
+    const currencyName = currency.getName()
+    const currencyCode = currency.getCode_asB64()
+    if (!currencyName) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Currency protobuf missing required field `name`.',
```

Comment:
if we use this many more times, i think it's probably worth adding a simple formatting function to reuse instead of typing this this out everytime, ie

`missingFieldMessage(fieldName: string)`